### PR TITLE
[Doc] Add missing mock import to docs `conf.py`

### LINKED ISSUE #6853: [Doc]: Sampling page is no longer showing up
### 📚 The doc issue

https://docs.vllm.ai/en/stable/dev/sampling_params.html
https://docs.vllm.ai/en/latest/dev/sampling_params.html

Screenshot 2024-07-26 at 4 36 36 PM

### Suggest a potential alternative/fix

Probably a typo in the source

### PR DESCRIPTION
**Pull request by hmellor**
Fixes the issue identified in https://github.com/vllm-project/vllm/pull/6600#issuecomment-2252739458

FIX #6853

### CODE REVIEW

**Comment by DarkLight1337** [comment]
Thanks for fixing this! Is there a way so that failures like https://readthedocs.org/projects/vllm/builds/25082568/ would fail the CI?

**Review by DarkLight1337 — APPROVED** [review]

**Comment by hmellor** [comment]
We could use some combination of [nitpicky mode](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-nitpicky) and [fail on warning](https://docs.readthedocs.io/en/stable/config-file/v2.html#sphinx-fail-on-warning)?

**Comment by DarkLight1337** [comment]
Seems like we can update `.readthedocs.yaml` with those options.

https://docs.readthedocs.io/en/stable/config-file/v2.html

**Comment by hmellor** [comment]
@DarkLight1337 I can't seem to get it to fail...

**Comment by DarkLight1337** [comment]
`python -m sphinx -T -W --keep-going -b html -d _build/doctrees -D language=en . $READTHEDOCS_OUTPUT/html`

I think it's because of `-W --keep-going`. Where did that come from?

**Comment by hmellor** [comment]
That is what `fail_on_warning` adds. It turns warnings into errors and adds keep going so that it doesn't crash at the first warning.

**Comment by DarkLight1337** [comment]
Oh, I didn't read the details of the CLI flags and thought it meant to skip the warnings. Hmm, in that case the build should have failed...

**Comment by hmellor** [comment]
Oh it did fail in AWS https://buildkite.com/vllm/ci-aws/builds/5709#0190ef62-e04f-4fb9-bfb2-bd29b30540e0

**Comment by DarkLight1337** [comment]
Nice! Let's put the fix back in this PR and get it merged then.

File: docs/source/conf.py [hunk 1]
```diff
@@ -93,7 +93,9 @@ def setup(app):
 
 
 # Mock out external dependencies here, otherwise the autodoc pages may be blank.
+nitpicky = True
 autodoc_mock_imports = [
+    # "aiohttp",
```
Comment (hmellor) [inline docs/source/conf.py:98]:
```suggestion
    "aiohttp",
```

**Comment by DarkLight1337** [comment]
Ok so it seems that we need nitpicky mode to catch missing mock imports, but doing so would cause a bunch of other errors. Let's fix the docs first and revisit the issue later.

**Pull request merged**